client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -0,0 +1,282 @@
+import { DataForm } from '@automattic/dataviews';
+import { useQuery, useMutation, useQueryClient, Query } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+	Modal,
+	ProgressBar,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState, useCallback } from 'react';
+import {
+	siteResetContentSummaryQuery,
+	resetSiteMutation,
+	siteResetStatusQuery,
+} from '../../app/queries';
+import Notice from '../../components/notice';
+import ContentInfo from './content-info';
+import type { Site, SiteContent, SiteResetStatus } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+import './style.scss';
+
+interface ErrorResponse {
+	code: string;
+	message: string;
+}
+
+type SiteResetFormData = {
+	domain: string;
+};
+
+function ErrorContent( { message, onClose }: { message: string; onClose: () => void } ) {
+	return (
+		<VStack spacing={ 6 }>
+			<Text>{ message }</Text>
+			<HStack justify="flex-end">
+				<Button variant="primary" onClick={ onClose }>
+					{ __( 'Ok' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+function InProgressContent( { progress }: { progress: number | undefined } ) {
+	const progressValue = progress ? progress * 100 : 10;
+	return (
+		<VStack spacing={ 4 }>
+			<ProgressBar className="reset-site-modal-progress-bar" value={ progressValue } />
+			<Text>{ __( "We're resetting your site. We'll email you once it's ready." ) }</Text>
+		</VStack>
+	);
+}
+
+function SiteResetContent( {
+	siteContent,
+	siteDomain,
+	isBusy,
+	onSubmit,
+	onClose,
+}: {
+	site: Site;
+	siteContent: SiteContent;
+	siteDomain: string;
+	isBusy: boolean;
+	onSubmit: () => void;
+	onClose: () => void;
+} ) {
+	const [ formData, setFormData ] = useState< SiteResetFormData >( {
+		domain: '',
+	} );
+
+	const fields: Field< SiteResetFormData >[] = [
+		{
+			id: 'domain',
+			label: __( 'Type the site domain to confirm' ),
+			type: 'text' as const,
+			description: sprintf(
+				/* translators: %s: site domain */
+				__( 'The site domain is: %s' ),
+				siteDomain
+			),
+		},
+	];
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		onSubmit();
+	};
+
+	return (
+		<VStack spacing={ 6 }>
+			<Notice variant="warning" density="medium">
+				<Text>
+					{ createInterpolateElement(
+						__(
+							'Before resetting your site, consider <link>exporting your content as a backup</link> ↗.'
```

Comment:
This notice will benefit from using the new notice, as soon as we can. It may already be available? dhW7zYBamXJIeJIE3wy5f4-fi-111_840

Reply:
Yep, added in a later commit. See https://github.com/Automattic/wp-calypso/pull/103723#issuecomment-2915193655.